### ISSUE TITLE
Example of how to attach a disk from one vm to another.

### ISSUE THREAD

**Issue by Smithx10** [vmware/govmomi#3503]
While trying to attempt some automation,  we hit a snag where we could not figure out how to Add / Attach a vmdk from 1 vm to another.  

Is there some simple steps on how to do this?  I've had to no luck.  An example in the documentation would be helpful.

**Comment by dougm** [comment]
Hi @Smithx10 , are you looking to do this with govc or with the API?

**Comment by Smithx10** [comment]
I found it here:  Not sure if this is the only way,  but it worked.  Was a bit of  a strange to get a disk from a VM and then have to override its size to 0 to Add it to another machine.    

```go
		// Need FileOperation=="" to add an existing disk, see object.VirtualMachine.configureDevice
		disk.CapacityInKB = 0
		disk.CapacityInBytes = 0
		if err = vm.AddDevice(ctx, d); err != nil {
			t.Error(err)
		}
```
https://github.com/vmware/govmomi/blob/314c6523b1f9bb723250023d14e683ac90e52922/simulator/virtual_machine_test.go#L513

**Comment by dougm** [comment]
Yes, that is a bit strange, you can use the spec+method underneath `AddDevice` to avoid that. But, you may want to considering using FCD (First Class Disk) instead. Recent example went in:
https://github.com/vmware/govmomi/blob/main/govc/disk/attach.go#L96
https://github.com/vmware/govmomi/blob/main/govc/disk/detach.go#L65
